server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/InternalRange.java

```diff
@@ -157,17 +157,21 @@ private static String generateKey(double from, double to, DocValueFormat format)
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            if (out.getVersion().onOrAfter(Version.V_7_17_1)) {
-                out.writeOptionalString(key);
-            } else {
+            if (out.getVersion().equals(Version.V_8_0_0)) {
                 out.writeString(key == null ? generateKey(from, to, format) : key);
+            } else {
+                if (out.getVersion().onOrAfter(Version.V_7_17_1)) {
+                    out.writeOptionalString(key);
+                } else {
+                    out.writeString(key == null ? generateKey(from, to, format) : key);
+                }
             }
             out.writeDouble(from);
-            if (out.getVersion().onOrAfter(Version.V_7_17_0)) {
+            if (out.getVersion().onOrAfter(Version.V_7_17_0) && out.getVersion().before(Version.V_8_2_0)) {
```

Comment:
I think you should move the v_8_2_0 thing to a different PR. it'll save a byte which is great but it makes the PR hard to backport.